Add 3rd party dependencies guidance (#33512)
dotnet/docs: styleguide/3rdPartyDependencies.md

> # 3rd party dependencies

Reviewer: ```suggestion
# Third-party dependencies
```

> When code samples take dependencies, our goal is that readers can use the material without additional research. This requires restricting dependencies to licenses that are well-established and understood for both open-source and commercial offerings. We categorize dependencies according to the licensing schemes here:

Reviewer: ```suggestion
When code samples take dependencies, our goal is that readers can use the material without additional research. This requires restricting dependencies to licenses that are well established and understood for both open-source and commercial offerings. We categorize dependencies according to the licensing schemes here:
```

> - **Open-source** : Rely on the [Open Source Initiative (OSI)](https://opensource.org/) to provide definitions for us. Offerings must meet the [definition](https://opensource.org/osd) provided by OSI, including using an [OSI approved license](https://opensource.org/licenses). Any component whose license is not an OSI approved license is considered a commercial offering. Prefer OSS projects that are members of any of the [OSS foundations that Microsoft is part of](https://opensource.microsoft.com/ecosystem/). Prefer permissive licenses for libraries. GPL or similar licenses are allowed for tools, OSes, and similar components.

Reviewer: ```suggestion
- **Open-source** : Rely on the [Open Source Initiative (OSI)](https://opensource.org/) to provide definitions for us. Offerings must meet the [definition](https://opensource.org/osd) provided by OSI, including using an [OSI approved license](https://opensource.org/licenses). Any component whose license is not an OSI approved license is considered a commercial offering. Prefer OSS projects that are members of any of the [OSS foundations that Microsoft is part of](https://opensource.microsoft.com/ecosystem/). Prefer permissive licenses for libraries. GPL or similar licenses are allowed for tools, OSs, and similar components.
```

Reviewer: See https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=42256.

> - **Commercial**: Samples may take dependencies on commercial offerings if readers can learn from our content without unnecessary additional costs. Typically, this means some form of a community edition, or a free trial sufficient for its use in content. This may be a form of dual-license, or tiered license.

Reviewer: ```suggestion
- **Commercial**: Samples may take dependencies on commercial offerings if readers can learn from our content without unnecessary additional costs. Typically, this means a community edition or free trial sufficient for its use in content is available. A commercial license scheme may be a form of dual license or tiered license.
```

> - **Dual-licensed**: Readers choose either license based on their needs. For example, if the offering has an OSS and commercial license, readers can  choose between them.

Reviewer: ```suggestion
- **Dual licensed**: Readers choose either of two licenses based on their needs. For example, if the offering has an OSS and commercial license, readers can choose between them.
```

> - **Tiered licensed**: Readers are expected to use the license tier that corresponds to their characteristics. For example, tiers may be available for students, hobbyists, or companies with defined revenue  thresholds. For offerings with tiered licenses, strive to limit our use in tutorials to the features available in the lowest tier. This enables the widest audience for the article. [Visual Studio](https://visualstudio.com) and [IdentityServer](https://www.identityserver.com/) are examples of this license type used in dotnet docs.

Reviewer: ```suggestion
- **Tiered licensed**: Readers are expected to use the license tier that corresponds to their characteristics. For example, tiers may be available for students, hobbyists, or companies with defined revenue thresholds. For offerings with tiered licenses, we strive to limit our use in tutorials to the features available in the lowest tier. This enables the widest audience for the article. [Visual Studio](https://visualstudio.com) and [IdentityServer](https://www.identityserver.com/) are examples of this license type used in dotnet docs.
```

> In general, we prefer taking dependencies on licensed components in the order of the listed categories. In cases where the category may not be well known, we will document the category so that readers understand the choice that they are making using that dependency.

Reviewer: ```suggestion
In general, we prefer taking dependencies on licensed components in the order of the listed categories. In cases where the category may not be well known, we'll document the category so that readers understand the choice that they're making by using that dependency.
```
